### ISSUE TITLE
Text Emotion Detection

### ISSUE THREAD

**Issue by Madhu0-2** [abhisheks008/ML-Crate#662]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Text Emotion Detection
</br>:red_circle: **Aim** : The aim is to predict emotion based on the text provided.
</br>:red_circle: **Dataset** : https://www.kaggle.com/code/khuzaimaaziz/text-emotion-detection-on-emotion-dataset/notebook
</br>:red_circle: **Approach** : Streamlit app.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : M. Madhu Sravanthi
- GitHub Profile Link :  https://github.com/Madhu0-2
- Participant ID (If not, then put NA) : NA
- Approach for this Project : Project Includes multiple steps:

   1. Data collection

   2. Data preparation

   3. Feature Engineering

   4. Training and testing the model

    5. Model deployment

    I will make a pickle file of the Fitted ML model and then deploy it to Streamlit Python Framework to display the text box where it 
    takes the text input and predict the emotion based on the text with accuracy. 

- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.) VSoC'24

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Madhu0-2** [comment]
Requesting to assign this issue to me : @Madhu0-2

**Comment by gaurimadan** [comment]
Hello Sir, I have been working on such ML projects such as recommender systems and face account prediction ,I am looking to work on NLP too. Please assign the issue to me .

**Comment by abhisheks008** [comment]
Hi @Madhu0-2 thanks for creating the issue. Are you planning to develop the models as well as the web app with the best fitted models developed by you?

**Comment by devil-90** [comment]
can you please assign me this project under vsoc

**Comment by divyansh-2707** [comment]
hello @abhisheks008 please assign this issue to me...

**Comment by fspzar123** [comment]
Full name : Filbert Shawn
GitHub Profile Link : https://github.com/fspzar123
Participant ID : NA

Approach for the Project : 

**Data Collection:**
Commonly used datasets include the [Emotion Dataset](https://www.kaggle.com/datasets/nelgiriyewithana/emotions/data) (e.g., Twitter data labeled with emotions) 

**Data Preprocessing:**
- Clean the text data by removing stop words, punctuation, and stemming and lemmatization.
- Tokenize the text into words or subwords.
- Convert the text into numerical representations using techniques like Bag of Words (BoW), Term Frequency-Inverse Document Frequency (TF-IDF).

**Model Selection:**
Traditional Machine Learning: Logistic Regression, SVM, Random Forest
Deep Learning:  RNNs, Long Short-Term Memory (LSTM) networks, Convolutional Neural Networks (CNNs)

**Model Training:**
- Split the dataset into training and testing sets.
- Train the model on the training data.
- Fine-tune hyperparameters and optimize the model.

**Evaluation:**
Evaluate the model using metrics like accuracy, precision, recall, F1 score, and confusion matrix.
Perform cross-validation to ensure the model generalizes well.

**Prediction:**
Use the trained model to predict emotions on new text data.


What is your participant role? SSOC'24

**Comment by abhisheks008** [comment]
> Full name : Filbert Shawn GitHub Profile Link : https://github.com/fspzar123 Participant ID : NA
> 
> Approach for the Project :
> 
> **Data Collection:** Commonly used datasets include the [Emotion Dataset](https://www.kaggle.com/datasets/nelgiriyewithana/emotions/data) (e.g., Twitter data labeled with emotions)
> 
> **Data Preprocessing:**
> 
> * Clean the text data by removing stop words, punctuation, and stemming and lemmatization.
> * Tokenize the text into words or subwords.
> * Convert the text into numerical representations using techniques like Bag of Words (BoW), Term Frequency-Inverse Document Frequency (TF-IDF).
> 
> **Model Selection:** Traditional Machine Learning: Logistic Regression, SVM, Random Forest Deep Learning: RNNs, Long Short-Term Memory (LSTM) networks, Convolutional Neural Networks (CNNs)
> 
> **Model Training:**
> 
> * Split the dataset into training and testing sets.
> * Train the model on the training data.
> * Fine-tune hyperparameters and optimize the model.
> 
> **Evaluation:** Evaluate the model using metrics like accuracy, precision, recall, F1 score, and confusion matrix. Perform cross-validation to ensure the model generalizes well.
> 
> **Prediction:** Use the trained model to predict emotions on new text data.
> 
> What is your participant role? SSOC'24

As this issue is opened by a contributor I can't assign it to others.
